modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpResponse.java

```diff
@@ -20,19 +20,21 @@
 package org.elasticsearch.http.netty4;
 
 import io.netty.handler.codec.http.DefaultFullHttpResponse;
+import io.netty.handler.codec.http.HttpHeaders;
 import io.netty.handler.codec.http.HttpResponseStatus;
+import io.netty.handler.codec.http.HttpVersion;
 import org.elasticsearch.common.bytes.BytesReference;
 import org.elasticsearch.http.HttpResponse;
 import org.elasticsearch.rest.RestStatus;
 import org.elasticsearch.transport.netty4.Netty4Utils;
 
 public class Netty4HttpResponse extends DefaultFullHttpResponse implements HttpResponse {
 
-    private final Netty4HttpRequest request;
+    private final HttpHeaders requestHeaders;
 
-    Netty4HttpResponse(Netty4HttpRequest request, RestStatus status, BytesReference content) {
-        super(request.nettyRequest().protocolVersion(), HttpResponseStatus.valueOf(status.getStatus()), Netty4Utils.toByteBuf(content));
-        this.request = request;
+    Netty4HttpResponse(HttpHeaders requestHeaders, HttpVersion version, RestStatus status, BytesReference content) {
```

Comment:
I figured if we release the request before using these parts (version and headers) it's a little nicer not to pass the full request around. Also, for non-pooled allocation (this includes all requests that aren't search or bulk) this makes the GC's life a little easier)